Design-Docs/General Design Doc.md

```diff
@@ -0,0 +1,131 @@
+  
+* Playback -> US Map (Drawer; Pink)
+  
+  The Playback is a button that once pressed will trigger the US Map Animation based on the type and speed options that were selected.
+  
+  * Secondary Action
+    
+    The playback button is pressed, the slider should slide over accordingly.
+
+#### The Animation
+
+From 1999 to 2016 there is data on the different types of Debt. The site will show that data on the map using the colors which 
+will change as the years go up or down. The color transitions should relatively smooth from year to year. The year should be animated as well, such as a scroll up down.
+
+
+### Animated Table 
+![Shit Tier Wireframe](Wireframes/Photos/Animated_Table.jpg "Basic Wireframe")
+The Animated table is the second medium that will be used to show the user the SDD. The Table will have the States as the rows and the Years as the columns. There will be a drop down that has the different types of debt and the population. Depending on what is selected in the drop down, the table will display that data. 
+
+To help the user read the data easier, table will color code the data in different ways. 
+* Highlights By Year Using Timed Intervals
+  * Color Coating much like the states
+  * Plain Yellow
+* Show and Hide States 
+  * Fade States In And Out as need to help user focus.
+  * Fuzzy Hide
+
+#### Component Breakdown
+(Note: Nested List Items Are Considered Child Elements)
+![Shit Tier Wireframe](Wireframes/Component_Breakdown/Animated_Table_CB.png)
+* Animated Table (Middle; Green)
+  * Table Header (Top; Blue)
+  * Table (Center; Yellow)
+    * Table Row (Middle Rect; Light Blue)
+      * Table Cell (Far Left; Peach)
+* Table Control Panel (Far Right; Red)
+  * Debt Type DD (Top; Orange)
+  * Presentation DD (2nd Top; Sea Green)
+  * Speed DD (3rd Top; Purple)
+  * Playback (Bottom; Pink)
+
+#### Component Relationship
+(Note: Some Relationships will not be displayed in the image, as concepts evolved over time)
+![Shit Tier Wireframe](Wireframes/Component_Relationship/Animated_Table_CR.png)
+* Debt Type DD -> Header; US Map (White; Top Right)
+
+  The Debt Type DD contains a list of the different type of debts, once one is chosen, the header will change to reflect the selection. The Animated Table will display the data of the new type selection.
+  
+* Presentation Type -> Table Row; Table Cell
+
+  The Presentation Type DD contains a list of the different coloring options the animated table can use highlight the data (see above). Once a selection is made, a new directive that adds the selected highlight behavoir will be added to the table cell, while the previous selection will be removed.  
+
+   Additionally, if the 'Color Code All' option which applies the color coding to the table all at once, is selected, then the Speed DD and Playback Components are hidden. (Note: The far right highlighting of the three columns represents what that would look like, but it would be apply throughout the table)
```

Comment:
Instead of hidden, can the elements be disabled so that they retain visibility in the layout?